lib/orogen/gen/project.rb

```diff
@@ -305,6 +305,22 @@ def find_typekit(typekit_name)
                 project.used_typekits.find { |tk| tk.name == typekit_name }
             end
 
+            # Returns all subdirectores that task-extens want to add
+            # This is used within the CMake-list generation to add
+            # custom targets to the build process
+            def additional_plugin_source_dirs
+                dirs = []
+                tasks.each do |name,task|
+                    task.extensions.each do |ext|
+                        next if !ext.respond_to? "build_subdirectories"
+                        ext.build_subdirectories.each do |elem|
```

Comment:
They are source directories, not build directories. And they call them "auto" (since they are in the code-generation part), e.g. each_auto_source_directory